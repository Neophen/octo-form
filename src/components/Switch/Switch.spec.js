import { shallowMount } from "@vue/test-utils";
import OSwitch from "./index";

let wrapper;

describe("OSwitch", () => {
  beforeEach(() => {
    wrapper = shallowMount(OSwitch);
  });

  it("is called", () => {
    expect(wrapper.name()).toBe("OSwitch");
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders input element by default", () => {
    expect(wrapper.contains("input")).toBeTruthy();
    expect(wrapper.classes()).toContain("octo-switch");
  });

  it("updates internal value when v-model is changed", done => {
    const newValue = "switch value";
    wrapper.setProps({ value: newValue });
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.newValue).toBe(newValue);
      done();
    });
  });

  it("emit input event when computedValue is set", () => {
    const newValue = "switch value";
    wrapper.vm.computedValue = newValue;
    const valueEmitted = wrapper.emitted()["input"][0];
    expect(wrapper.vm.newValue).toBe(newValue);
    expect(valueEmitted).toContainEqual(newValue);
  });

  it("method focus() gives focus to the input element", done => {
    wrapper.vm.$refs.input.focus = jest.fn();
    wrapper.vm.focus();
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$refs.input.focus).toHaveBeenCalled();
      done();
    });
  });
});
