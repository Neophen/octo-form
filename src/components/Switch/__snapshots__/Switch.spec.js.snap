// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSwitch render correctly 1`] = `
<label class="octo-switch is-rounded">
  <!----> <input type="checkbox" true-value="true" class="octo-switch__input" value="false"> <span class="octo-switch__check"></span> <span class="octo-switch__label is-right"></span></label>
`;
